src/main/java/software/amazon/awssdk/crt/io/Uri.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+package software.amazon.awssdk.crt.io;
+
+import java.nio.charset.StandardCharsets;
+import software.amazon.awssdk.crt.CRT;
+
+public class Uri {
+    static {
+        new CRT();
+    };
+
+    /**
+     * Returns a concatanation of an encoded base, and the uri path encoding of a
+     * string. This is the modified version of rfc3986 used by sigv4 signing.
+     */
+    public static String appendEncodingUriPath(String encoded, String path) {
+        return new String(
+                appendEncodingUriPath(encoded.getBytes(StandardCharsets.UTF_8), path.getBytes(StandardCharsets.UTF_8)),
+                StandardCharsets.UTF_8);
+    }
+
+    public static String encodeUriPath(String path) {
```

Comment:
/**
     * Returns the uri path encoding of the string. This is the modified version of rfc3986 used by sigv4 signing.
     * 
     * @param path The path to be encoded
     */